### ISSUE TITLE
Unsupported browser error on Postman desktop app (master issue)

### ISSUE THREAD

**Issue by shamasis** [a85/POSTMan-Chrome-Extension#7700]
**Describe the bug**
When using any feature that uses Postman App’s embedded browser, such as oauth2 token fetching, certain web pages display that the browser is unsupported.

**App information (please complete the following information):**
 - App Type: Native
 - Postman Version: 7.12 (below and all future version until fixed)
 - OS: all

**Additional context**
It is most likely because of outdated version of chromium packaged with the desktop app.

**Comment by numaanashraf** [comment]
More details:

Postman's embedded browser version, as of Postman `v7.12`, is `Chrome 66.0.3359.181`

**Comment by choowie** [comment]
This problem is blocking quite a few people. Version 66 was released in April 2018 and one year later, some serious security issues have been fixed in the latest version. Version 78 of Chrome fixes a vulnerability rated 9 out of 10 on the CVE scale.

Most companies are blocking older browsers and even Google complains when I authenticate with my google account "you are trying to sign in from a browser or app that doesn't allow us to keep you account secure".

When will a fix be released?

**Comment by yaimag** [comment]
Same thing is happening to me.....

**Comment by levyblue** [comment]
Same thing hanppening to me

**Comment by sarunprasath** [comment]
Is there any update on this bug ???

**Comment by Thainita** [comment]
I am currently facing the same problem, MacOS Catalia 10.15.2, Postman Version 7.16. Is there any updates on this matter or maybe any workarounds while we wait for the solution?

Thank you!

**Comment by Arametheus** [comment]
I have the same issue.  same setup as ThaiRodrigues7

**Comment by hexenmesser** [comment]
Hello. Experiencing the same issue (generating OAuth 2.0 for Gmail API using POSTMAN app) 
Are there any updates on this?

**Comment by tomejesus** [comment]
Any update on this? Also experiencing the same.

**Comment by wizofaus** [comment]
Likewise - cannot access my postman account at all because of this, have tried everything.
UPDATE: eventually had to uninstall and re-install latest version.

**Comment by SaschaHeyer** [comment]
I have the same behavior any updates?

**Comment by tkentdna** [comment]
Having the same issue.  Per the post by wizofaus (https://github.com/postmanlabs/postman-app-support/issues/7700#issuecomment-586788805), I uninstalled Postman and reinstalled the latest (v7.19.1), but the issue remains for me.

**Comment by lmflucca** [comment]
Hello,

I can access Postman to Google login, but I installed extension postman for Google Chrome, not desktop app.

Please, access this path to download extension postman [https://chrome.google.com/webstore/detail/postman/fhbjgbiflinjbdggehcddcbncdddomop/related?hl=en]

**Comment by Arametheus** [comment]
@lmflucca Yes that works but the issue is they are deprecating it and recommend to not use the extension. 

Only other way I have gotten it to work is create the bearer token else were like the google oath2 playground and then copy that into the bearer token field.  but of course you have to refresh it everytime it expires.

**Comment by perymimon** [comment]
the same thing when trying to make auth2.0 request from postman `Authorization/Oauth2.0 -> get new access token`

**Comment by titzu** [comment]
same for me. trying to make auth2.0 request from postman Authorization/Oauth2.0 -> get a new access token

**Comment by rmbryan71** [comment]
Is there a work-around that lets postman send traffic to sites that check browser version and don't allow old versions?

**Comment by elznr** [comment]
I have same problem. Can I get token elsewhere?

**Comment by dareales** [comment]
Until they fix this bug, you can use the chrome extension to get the token exclusively. Then paste the token into the desktop application to continue working.

**Comment by harryi3t** [comment]
Hi, 

We've pushed an update on our early access canary channel that should resolve this issue. Do try it out and let us know

You can get the canary app from https://www.getpostman.com/downloads/canary

**Comment by elznr** [comment]
I tried PostmanCanary and it does not work.

![C29479C3-F86F-465E-AC90-399C59AA89DF](https://user-images.githubusercontent.com/3471762/79425340-d6f4bb00-7fc1-11ea-8f4d-493c856fccfb.jpeg)

**Comment by harryi3t** [comment]
@elznr Thanks for quickly trying it out and letting us know.

Google is blocking all the embedded browsers, so even though we have updated the embedded browser to a relatively recent version (Chrome 78) it is still getting blocked.
For these cases, we are working on supporting the OAuth2 flows using the external Browsers. We will keep you posted on the updates.

Here's a glimpse of that upcoming feature:
![image](https://user-images.githubusercontent.com/5207331/79428578-fb618f00-7fe3-11ea-982f-503e5566d185.png)

**Comment by bjornvdakker** [comment]
Hi @harryi3t ,

When do you expect this feature to be launched in beta / production? Would be great if we could get this working!

**Comment by ryadaval** [comment]
> @lmflucca Yes that works but the issue is they are deprecating it and recommend to not use the extension.
> 
> Only other way I have gotten it to work is create the bearer token else were like the google oath2 playground and then copy that into the bearer token field. but of course you have to refresh it everytime it expires.

Thank you Arametheus. Your idea worked for me. For anyone else looking for the oath2 playground of google, here is the link. 
https://developers.google.com/oauthplayground/

Meanwhile, postman is still not working. At present the chrome plugin for postman is also not working.

**Comment by elznr** [comment]
Well today I updated to Postman v7.23.0 and now it is working for me. How come?

**Comment by rhaseven7h** [comment]
At least on 7.23.0 linux Ubuntu, it doesn't work today May 6th, 2020.

**Comment by rns4731** [comment]
Still not working with the latest postman.

**Comment by vikiCoder** [comment]
Hi everyone,

We've added an option in OAuth2 that allows you to authenticate using your system's default browser while generating OAuth 2.0 token.

To use this feature, select `OAuth 2.0` under `Authorization` tab and open `Get New Access Token` dialog box. You'll see a new checkbox `Authorize using browser` when `Authorization code` or `Implicit` grant type is selected. Selecting this option will allow you to use your external browser for authentication instead of using Postman's built-in browser.

However, selecting this option will automatically use the callback URL `https://oauth.pstmn.io/v1/callback`. You can't change this callback URL. This is necessary to return the generated token information back to the Postman app on successful authentication. So you'll have to whitelist this URL to be used as callback in your respective OAuth provider if required.

You can refer to [this link in our learning center](https://learning.postman.com/docs/postman/sending-api-requests/authorization/#requesting-an-oauth-20-token) for more details about generating OAuth 2.0 token. Here is a screenshot showing off this new feature. Please update to the latest Postman app (v7.24.0) and check it out. Let us know if you are still facing any issues related to this. Any feedbacks are welcome 🙂

![Capture](https://user-images.githubusercontent.com/16588032/81491083-6d598b00-92a7-11ea-937e-8b80f1f1cee5.PNG)

**Comment by rhaseven7h** [comment]
Just tested it to Oauth2 for a Web app in Auth0. Worked flawlessly.
💯 🥇

**Comment by MichaelBelgium** [comment]
Ok, thanks for that checkbox! I thought it wasn't working at first as google chrome blocked the popup that provides the auth response to go back to the postman app, so keep your eyes on blocked popups in case you're waiting 20 minutes for postman to get that auth response 😂

**Comment by arlemi** [comment]
@MichaelBelgium Thanks for reporting it, we're tracking adding this to the docs [here](https://github.com/postmanlabs/postman-docs/issues/2384), if you feel like giving it a go our docs are completely open source. 😀

**Comment by topshef** [comment]
What is the correct Auth URL for using GCP?
I am using https://ssh.cloud.google.com/ then preview at something like https://8080-dot-XXXXXX-dot-devshell.appspot.com/

Sorry if this is a noddy question or I'm in the wrong place! 

Ref: https://github.com/postmanlabs/postman-app-support/issues/7700#issuecomment-626273229 

Thanks!

**Comment by arlemi** [comment]
Hey @topshef , that info should be provided by GCP when you create the credentials on your developer console. If you're having issues finding them please ask the question on our [community forum](https://community.getpostman.com/) since this isn't really a problem with the app. :)

**Comment by topshef** [comment]
> Hey @topshef , that info should be provided by GCP when you create the credentials on your developer console. If you're having issues finding them please ask the question on our [community forum](https://community.getpostman.com/) since this isn't really a problem with the app. :)

Thanks a lot, it's posted now via community https://community.postman.com/t/giving-access-to-gcp-localhost/15340

**Comment by jprantza** [comment]
Hi, I see that #7720 was closed as a duplicate of #7700, but I'm failing to see how this helps with the issue identified in #7720.

The issue identified in #7720 is that we can't log into the Postman app (and therefore use workspaces and collections) because Postman auth is using Google Chrome, which is blocked by corporate. The fix in #7700 allows me to use a different browser for OAuth requests, but how do I configure it so that it applies to the Postman "Sign In" as well?

**Comment by yolowex** [comment]
I am using postman on Linux, I tried Firefox, Chromium & Falkon, but postman has no awareness of the browsers I'm using (or browsers I have) at all.
it can't even open the log in page on the browser.

you should probably let users login from the app itself? 
I don't see how it could be problamatic if you do so.

Also, expecting users to install certain browsers just to "Log in" in postman is disrespectful.
